internal/controllers/machinedeployment/machinedeployment_rolling.go

```diff
@@ -293,3 +299,25 @@ func (r *Reconciler) scaleDownOldMachineSetsForRollingUpdate(ctx context.Context
 
 	return totalScaledDown, nil
 }
+
+// cleanupDisableMachineCreateAnnotation will remove the disable machine create annotation from new MachineSets that were created during reconcileOldMachineSetsOnDelete.
+func (r *Reconciler) cleanupDisableMachineCreateAnnotation(ctx context.Context, newMS *clusterv1.MachineSet) (*clusterv1.MachineSet, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	if newMS.Annotations != nil {
```

Comment:
this duplicates the removal code. Also it would run twice in the onDelete case so we should be able to remove it there.

Maybe also move it to the top of this function.

Caveat: with doing this it won't be possible (anymore) to manually set this annotation on a MS if it is the new machineset for the normal rollout type. But IMHO this should be okay, because the MD controller owns the MS.

Reply:
It will never fully run twice since its checking the annotation on the MS if one run removes it the second one shouldn't run because the annotation isn't there. I can add a check before calling the function to check the annotation exists if that helps?

Reply:
my point is: I think we don't need it at ondelete anymore / we only need this code once :-)